src/main/telemetry/srxl.c

```diff
@@ -160,6 +162,180 @@ bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
+    uint8_t hdopBcd, gpsFlags;
+
+    if (!featureIsEnabled(FEATURE_GPS) || !STATE(GPS_FIX) || gpsSol.numSat < 6) {
+        return false;
+    }
+
+    // lattitude
+    GPStoDDDMM_MMMM(gpsSol.llh.lat, &coordinate);
+    latitudeBcd  = (dec2bcd(coordinate.dddmm) << 16) | dec2bcd(coordinate.mmmm);
+
+    // longitude
+    GPStoDDDMM_MMMM(gpsSol.llh.lon, &coordinate);
+    longitudeBcd = (dec2bcd(coordinate.dddmm) << 16) | dec2bcd(coordinate.mmmm);
+
+    // altitude (low order)
+    altitudeLo = ABS(gpsSol.llh.altCm) / 10;
+    altitudeLoBcd = dec2bcd(altitudeLo % 100000);
+
+    // Ground course
+    groundCourseBcd = dec2bcd(gpsSol.groundCourse);
+
+    // HDOP
+    hdop = gpsSol.hdop / 10;
+    hdop = (hdop > 99) ? 99 : hdop;
+    hdopBcd = dec2bcd(hdop);
+
+    // flags
+    gpsFlags = (gpsSol.llh.lat > 0)   ? (0x00     | GPS_FLAGS_IS_NORTH_BIT) : gpsFlags;
+    gpsFlags = (gpsSol.llh.lon > 0)   ? (gpsFlags | GPS_FLAGS_IS_EAST_BIT) : gpsFlags;
+    gpsFlags = (gpsSol.llh.altCm < 0) ? (gpsFlags | GPS_FLAGS_NEGATIVE_ALT_BIT) : gpsFlags;
+
+    gpsFlags = gpsFlags | GPS_FLAGS_GPS_DATA_RECEIVED_BIT;
+    gpsFlags = gpsFlags | GPS_FLAGS_GPS_FIX_VALID_BIT;
+    gpsFlags = gpsFlags | GPS_FLAGS_3D_FIX_BIT;
+
+    if (gpsSol.llh.lon / GPS_DEGREES_DIVIDER > 99) {
+        gpsFlags |= GPS_FLAGS_LONGITUDE_GREATER_99_BIT;
+    }
+
+    // SRXL frame
+    sbufWriteU8(dst, SRXL_FRAMETYPE_GPS_LOC);
+    sbufWriteU8(dst, SRXL_FRAMETYPE_SID);
+    sbufWriteU16(dst, altitudeLoBcd);
+    sbufWriteU32(dst, latitudeBcd);
+    sbufWriteU32(dst, longitudeBcd);
+    sbufWriteU16(dst, groundCourseBcd);
+    sbufWriteU8(dst, hdopBcd);
+    sbufWriteU8(dst, gpsFlags);
+
+    return true;
+}
+
+/*
+typedef struct
+{
+   UINT8   identifier;                      // Source device = 0x17
+   UINT8   sID;                             // Secondary ID
+   UINT16  speed;                           // BCD, knots, format 3.1
+   UINT32  UTC;                             // BCD, format HH:MM:SS.S, format 6.1
+   UINT8   numSats;                         // BCD, 0-99
+   UINT8   altitudeHigh;                    // BCD, meters, format 2.0 (High bits alt)
+} STRU_TELE_GPS_STAT;
+*/
+
+#define STRU_TELE_GPS_STAT_EMPTY_FIELDS_COUNT 6
+
+bool srxlFrameGpsStat(sbuf_t *dst, timeUs_t currentTimeUs)
+{
+    UNUSED(currentTimeUs);
+    uint32_t timeBcd;
+    uint16_t speedKnotsBcd, speedTmp;
+    uint8_t numSatBcd, altitudeHighBcd;
+    dateTime_t dt;
+    bool timeProvided = false;
+
+    if (!featureIsEnabled(FEATURE_GPS) || !STATE(GPS_FIX) || gpsSol.numSat < 6) {
+        return false;
+    }
+
+    // Number of sats and altitude (high bits)
+    numSatBcd = (gpsSol.numSat > 99) ? dec2bcd(99) : dec2bcd(gpsSol.numSat);
+    altitudeHighBcd = dec2bcd(gpsSol.llh.altCm / 100000);
+
+    // Speed (knots)
+    speedTmp = gpsSol.groundSpeed * 1944 / 1000;
+    speedKnotsBcd = (speedTmp > 9999) ? dec2bcd(9999) : dec2bcd(speedTmp);
+
+    // RTC
+    #ifdef USE_RTC_TIME
```

Comment:
Per the coding guidelines compiler directives should be left justified.

Reply:
Sure, forgot about this one.